src/sidebar/components/test/group-list-item-out-of-scope-test.js

```diff
@@ -1,106 +1,98 @@
 'use strict';
 
-const angular = require('angular');
+const { mount } = require('enzyme');
+const preact = require('preact');
+const { createElement } = require('preact');
 const proxyquire = require('proxyquire');
-const util = require('../../directive/test/util');
 
 const { events } = require('../../services/analytics');
 
-describe('groupListItemOutOfScope', () => {
+describe('GroupListItemOutOfScope', () => {
   let fakeAnalytics;
   let fakeGroupListItemCommon;
+  let GroupListItemOutOfScope;
+
+  const fakeGroup = {
+    id: 'groupid',
+    links: {
+      html: 'https://hypothes.is/groups/groupid',
+    },
+    logo: 'dummy://hypothes.is/logo.svg',
+    organization: { name: 'org' },
+  };
+
+  // Click on the item to expand or collapse it.
+  const toggle = wrapper =>
+    wrapper
+      .find('div')
+      .first()
+      .simulate('click');
 
   before(() => {
     fakeGroupListItemCommon = {
       orgName: sinon.stub(),
       trackViewGroupActivity: sinon.stub(),
     };
 
-    // Return groupListItemOutOfScope with groupListItemCommon stubbed out.
-    const groupListItemOutOfScope = proxyquire(
-      '../group-list-item-out-of-scope',
-      {
-        '../util/group-list-item-common': fakeGroupListItemCommon,
-        '@noCallThru': true,
-      }
-    );
+    GroupListItemOutOfScope = proxyquire('../group-list-item-out-of-scope', {
+      // Use same instance of Preact module in tests and mocked module.
+      // See https://robertknight.me.uk/posts/browserify-dependency-mocking/
+      preact,
```

Comment:
The need to pass through `preact` here is due to a rather confusing aspect of proxyquire and the reason I think we should replace it with some other means of mocking. The issue is not specific to Preact/React, but it comes up with any library that relies on certain `instanceof` checks or global state internally. This included Angular in the past.

Reply:
Hmmm, this _is_ a bit fiddly, eh? Is there any way to put this in another layer in the long term? Or does it absolutely need to be done locally in each test module?

Reply:
My preferred solution is to replace Proxyquire with a different method of mocking that doesn't have this problem. That's on my plate to do in the near future.